pyproject.toml

```diff
@@ -0,0 +1,4 @@
+#pyproject.toml
+[build-system]
+# Minimum requirements for the build system to execute.
+requires = ['setuptools>=45.0', 'wheel', 'cython>=0.29', 'oldest-supported-numpy', 'mpi4py==3.1.1']
```

Comment:
pip will still likely install a temporary version of mpi4py during the build regardless @kejacobson. This is how pip handles building cython packages. In general we pin against older versions of mpi4py because they are always forward compatible if the users system mpi4py is newer. I'd recommend keeping the mpi4py version number as was, since we've seen issues in the past with pip trying to build codes against newer versions mpi4py than the user is running

Reply:
Ok I changed this back to `mpi4py==3.1.1` then. I thought there was a reason that was the convention in TACS

Reply:
Interesting, it didn't seem forward compatible when I tried to build it. My pip failed when building mpi4py 3.1.1 with ==, but found my existing 3.1.4 with the >= change.

Reply:
Weird. Do you have the failure message?